gamemodes/terrortown/entities/entities/ttt_logic_role.lua

```diff
@@ -24,35 +26,46 @@ function ENT:KeyValue(key, value)
 			value = _G[value] or value
 		end
 
-		self.Role = tonumber(value)
+		self.checkingRole = tonumber(value)
 
-		if not self.Role then
+		if not self.checkingRole then
 			ErrorNoHalt("ttt_logic_role: bad value for Role key, not a number\n")
 
-			self.Role = ROLE_NONE
+			self.checkingRole = ROLE_NONE
```

Comment:
Wouldn't this break compatibility?

Reply:
not sure which addon would modify this entity

Reply:
Would be cool if @TheNickSkater could test this with DamageLogs as that add-on does somethings with role data :D